lib/features/auth/domain/auth_bloc.dart

```diff
@@ -0,0 +1,172 @@
+import 'dart:async';
+
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_facebook_auth/flutter_facebook_auth.dart';
+
+part 'auth_event.dart';
+
+part 'auth_state.dart';
+
+class AuthBloc extends Bloc<AuthEvent, AuthState> {
+  AuthBloc() : super(AuthInitial()) {
+    on<RegisterEvent>(_onRegister);
+    on<LoginEvent>(_onLogin);
+    on<LogoutEvent>(_onLogout);
+    on<GoogleEvent>(_handleGoogleSignIn);
+    on<FacebookEvent>(_logInWithFacebook);
+    on<ShowSignUpScreenEvent>((event, emit) {
+      emit(SignUpScreenState());
+    });
+    on<ShowAuthScreenEvent>((event, emit) {
+      emit(AuthScreenState());
+    });
+
+  }
+
+  Stream<AuthState> mapEventToState(AuthEvent event) async* {
+    if (event is LoginEvent) {
+      yield AuthLoadingState();
+      try {
+        yield AuthLoadedState();
+      } catch (e) {
+        yield AuthErrorState(e.toString());
+      }
+    } else if (event is ShowSignUpScreenEvent) {
+      yield SignUpScreenState();
+    } else if (event is GoogleEvent) {
+    } else if (event is FacebookEvent) {
+    }
+  }
+
+  final FirebaseAuth auth = FirebaseAuth.instance;
+
+  Future<void> _onRegister(RegisterEvent event, Emitter<AuthState> emit) async {
+    emit(AuthLoadingState());
+    try {
+      if (event.email.isEmpty || event.password.isEmpty || event.name.isEmpty) {
+        emit(AuthErrorState('Please fill out all fields'));
+        return;
+      }
+
+      final userCredential = await FirebaseAuth.instance.createUserWithEmailAndPassword(
+        email: event.email,
+        password: event.password,
+      );
+
+      await userCredential.user?.updateProfile(displayName: event.name);
```

Comment:
Why using updateProfile instead of updateDisplayName ?

Reply:
Done